healthcare/healthcare/doctype/nursing_task/nursing_task.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021, healthcare and contributors
+# For license information, please see license.txt
+import frappe
+from frappe.model.document import Document
+
+
+class NursingTask(Document):
+	@classmethod
+	def create_nursing_tasks_from_template(cls, template_name=None, dt=None, dn=None, append=False):
+		tasks = frappe.get_list(
+			'Nursing Checklist Template Task',
+			filters={'parent': template_name},
+			fields=['*'],
+		)
+		if append:
+			NursingTask.create_nursing_tasks(dt, dn, tasks)
+		else:
+			filters = {
+				'reference_doctype': dt,
+				'reference_name': dn,
+			}
+			existing_tasks = frappe.get_list('Nursing Task', filters=filters)
+			if existing_tasks:
+				return
+			NursingTask.create_nursing_tasks(dt, dn, tasks)
+
+	@classmethod
+	def create_nursing_tasks(cls, dt=None, dn=None, tasks=None):
+		for task in tasks:
+			options = {
+				'doctype': 'Nursing Task',
+				'status': 'Requested',
+				'reference_doctype': dt,
+				'reference_name': dn,
+				'activity': task.activity,
+				'mandatory': task.mandatory,
+				'task_doctype': task.task_doctype,
```

Comment:
No mapping of dates from template to task?

The From Date should be set relative to Reference Doctype's transaction date. And End Date should be calculated as From Date + Duration

Check how [https://docs.erpnext.com/docs/v13/user/manual/en/human-resources/employee-onboarding#31-employee-onboarding-template](https://docs.erpnext.com/docs/v13/user/manual/en/human-resources/employee-onboarding#31-employee-onboarding-template works) in ERPNext.

Reply:
Added a time offset field to set the requested_start_time